src/__tests__/funders.int.test.ts

```diff
@@ -0,0 +1,159 @@
+import request from 'supertest';
+import { app } from '../app';
+import {
+	createOrUpdateFunder,
+	loadFunder,
+	loadTableMetrics,
+} from '../database';
+import { expectTimestamp } from '../test/utils';
+import {
+	mockJwt as authHeader,
+	mockJwtWithAdminRole as adminUserAuthHeader,
+} from '../test/mockJwt';
+
+const agent = request.agent(app);
+
+describe('/funders', () => {
+	describe('GET /', () => {
+		it('requires authentication', async () => {
+			await agent.get('/funders').expect(401);
+		});
+
+		it('returns an empty bundle when no data is present', async () => {
+			await agent.get('/funders').set(authHeader).expect(200, {
+				entries: [],
+				total: 0,
+			});
+		});
+
+		it('returns all funders present in the database', async () => {
+			await createOrUpdateFunder({
+				shortCode: 'theFundFund',
+				name: 'The Fund Fund',
+			});
+			await createOrUpdateFunder({
+				shortCode: 'theFoundationFoundation',
+				name: 'The Foundation Foundation',
+			});
+
+			const response = await agent.get('/funders').set(authHeader).expect(200);
+			expect(response.body).toEqual({
```

Comment:
A few things going on

1. I generally like being able to organize tests in `Arrange` `Act` `Assert` -- in some cases the check of a response is mixed in with other checks, and being able to decouple it visually from the `Act` section allows full flexibility for organizing the assertions.  That isn't the case here but still mentioning it

2. Using `.expect(status, [object])` is using supertest's equality checks, which is less powerful than jest's.  You're right that `.expect(status, [fn])` would allow us to have something like:


```
.expect(200, (response) => {
   expect(response).toMatch({
      ...
   });
}).
```

But I'm not convinced that is actually more legible than

```
result = await agent.get(...);

expect(result.body).toEqual(...);
```

There's a third option here which is to remove `.expect(status)` entirely and ONLY use jest for assertion (e.g. add an `expect(response.status).toBe(200)`.  This would be cleaner from an Arrange Act Assertion organization, and would mean we aren't dealing with two types of assertion engine.

Thoughts?

Reply:
Awesome, thanks for that context! The way it is makes sense now, and I don't think we need to change our patterns here.